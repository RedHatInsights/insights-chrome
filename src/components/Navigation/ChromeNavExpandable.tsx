import React from 'react';
import { NavExpandable } from '@patternfly/react-core/dist/dynamic/components/Nav';
import ChromeNavItemFactory from './ChromeNavItemFactory';
import { ChromeNavExpandableProps } from '../../@types/types';

const ChromeNavExpandable = ({ title, routes, active, isHidden, id }: ChromeNavExpandableProps) => {
  if (isHidden || routes.filter(({ appId, expandable }) => expandable || !!appId).length === 0) {
    return null;
  }

  const quickStartHighlightId = title.replace(/\s/g, '-');
  return (
    <NavExpandable id={id} isExpanded={active} isActive={active} title={title} data-quickstart-id={quickStartHighlightId}>
      {routes.map((item, index) => (
        <ChromeNavItemFactory key={index} {...item} />
      ))}
    </NavExpandable>
  );
};

export default ChromeNavExpandable;
