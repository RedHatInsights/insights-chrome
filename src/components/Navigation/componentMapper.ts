import ChromeNavGroup from './ChromeNavGroup';
import ChromeNavExapandable from './ChromeNavExapandable';
import ChromeNavItem from './ChromeNavItem';
import DynamicNav from './DynamicNav';

const componentMapper = {
  group: ChromeNavGroup,
  expandable: ChromeNavExapandable,
  item: ChromeNavItem,
  dynamicNav: DynamicNav,
};

export default componentMapper;
