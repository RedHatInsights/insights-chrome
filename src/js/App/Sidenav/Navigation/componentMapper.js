import ChromeNavGroup from './ChromeNavGroup';
import ChromeNavExapandable from './ChromeNavExapandable';
import ChromeNavItem from './ChromeNavItem';

const componentMapper = {
  group: ChromeNavGroup,
  expandable: ChromeNavExapandable,
  item: ChromeNavItem,
};

export default componentMapper;
