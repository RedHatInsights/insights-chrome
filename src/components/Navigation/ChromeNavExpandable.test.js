import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import createMockStore from 'redux-mock-store';
import NavContext from './navContext';
import ChromeNavExpandable from './ChromeNavExpandable';
import componentMapper from './componentMapper';

const NavContextWrapper = ({
  store,
  providerValue = {
    onLinkClick: jest.fn(),
    componentMapper,
  },
  children,
  initialEntries,
}) => (
  <MemoryRouter initialEntries={initialEntries}>
    <Provider store={store}>
      <NavContext.Provider value={providerValue}>{children}</NavContext.Provider>
    </Provider>
  </MemoryRouter>
);

describe('ChromeNavExpandable', () => {
  const mockStore = createMockStore();
  const expandableTitle = 'Foo';
  const testProps = {
    appId: 'testModule',
    href: '/foo',
    title: expandableTitle,
    routes: [{ appId: 'bar', title: 'title', href: '/foo/bar' }],
    id: 'test-id',
  };
  const store = mockStore({
    chrome: {
      activeModule: 'testModule',
      moduleRoutes: [],
      modules: {
        testModule: {},
      },
    },
  });

  test('should not render nav item expandable', () => {
    const { queryAllByText, container } = render(
      <NavContextWrapper store={store}>
        <ChromeNavExpandable isHidden {...testProps} />
      </NavContextWrapper>
    );
    expect(container).toMatchSnapshot();
    expect(queryAllByText(expandableTitle)).toHaveLength(0);
  });

  test('should render nav item expandable', () => {
    const { queryAllByText, container } = render(
      <NavContextWrapper store={store}>
        <ChromeNavExpandable {...testProps} />
      </NavContextWrapper>
    );
    expect(container).toMatchSnapshot();
    expect(queryAllByText(expandableTitle)).toHaveLength(1);
  });

  test('should render nav item expandable with items', () => {
    const { queryAllByText, container } = render(
      <NavContextWrapper store={store}>
        <ChromeNavExpandable
          {...testProps}
          routes={[
            {
              appId: 'testModule',
              href: '/foo',
              title: 'Sub Item',
            },
          ]}
        />
      </NavContextWrapper>
    );
    expect(container).toMatchSnapshot();
    expect(queryAllByText(expandableTitle)).toHaveLength(1);
    expect(queryAllByText('Sub Item')).toHaveLength(1);
  });
});
