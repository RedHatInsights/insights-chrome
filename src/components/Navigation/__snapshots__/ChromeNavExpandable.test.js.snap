// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChromeNavExpandable should not render nav item expandable 1`] = `<div />`;

exports[`ChromeNavExpandable should render nav item expandable 1`] = `
<div>
  <li
    class="pf-v5-c-nav__item pf-m-expandable"
    data-ouia-component-id="OUIA-Generated-NavExpandable-1"
    data-ouia-component-type="PF5/NavExpandable"
    data-ouia-safe="true"
    data-quickstart-id="Foo"
  >
    <button
      aria-expanded="false"
      class="pf-v5-c-nav__link"
      id="test-id"
    >
      Foo
      <span
        class="pf-v5-c-nav__toggle"
      >
        <span
          class="pf-v5-c-nav__toggle-icon"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </span>
      </span>
    </button>
    <section
      aria-labelledby="test-id"
      class="pf-v5-c-nav__subnav"
      hidden=""
    >
      <ul
        class="pf-v5-c-nav__list"
        role="list"
      >
        <li
          class="pf-v5-c-nav__item chr-c-navigation__item"
          data-ouia-component-id="title"
          data-ouia-component-type="PF5/NavItem"
          data-ouia-safe="true"
        >
          <a
            class="pf-v5-c-nav__link chr-c-navigation__item"
            data-quickstart-id="foo_bar"
            data-testid="router-link"
            href="/foo/bar"
            itemid="/foo/bar"
          >
            Title
             
          </a>
        </li>
      </ul>
    </section>
  </li>
</div>
`;

exports[`ChromeNavExpandable should render nav item expandable with items 1`] = `
<div>
  <li
    class="pf-v5-c-nav__item pf-m-expandable"
    data-ouia-component-id="OUIA-Generated-NavExpandable-2"
    data-ouia-component-type="PF5/NavExpandable"
    data-ouia-safe="true"
    data-quickstart-id="Foo"
  >
    <button
      aria-expanded="false"
      class="pf-v5-c-nav__link"
      id="test-id"
    >
      Foo
      <span
        class="pf-v5-c-nav__toggle"
      >
        <span
          class="pf-v5-c-nav__toggle-icon"
        >
          <svg
            aria-hidden="true"
            class="pf-v5-svg"
            fill="currentColor"
            height="1em"
            role="img"
            viewBox="0 0 256 512"
            width="1em"
          >
            <path
              d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
            />
          </svg>
        </span>
      </span>
    </button>
    <section
      aria-labelledby="test-id"
      class="pf-v5-c-nav__subnav"
      hidden=""
    >
      <ul
        class="pf-v5-c-nav__list"
        role="list"
      >
        <li
          class="pf-v5-c-nav__item chr-c-navigation__item"
          data-ouia-component-id="Sub Item"
          data-ouia-component-type="PF5/NavItem"
          data-ouia-safe="true"
        >
          <a
            class="pf-v5-c-nav__link chr-c-navigation__item"
            data-quickstart-id="foo"
            data-testid="router-link"
            href="/foo"
            itemid="/foo"
          >
            Sub Item
             
          </a>
        </li>
      </ul>
    </section>
  </li>
</div>
`;
